ldap_sync/exporter.py

```diff
@@ -147,6 +184,57 @@ def fetch_users_to_sync(session, required_property=None):
     )
 
 
+def fetch_groups_to_sync(session) -> List[GroupProxyType]:
+    """Fetch the groups who should be synced
+
+    :param session: The SQLAlchemy session to use
+
+    :returns: An iterable of `(Group, members)` ResultProxies.
+    """
+    return (
+        # Grab all users with the required property
+        Group.q
+        # uids of the members of the group
+        .add_column(select([func.array_agg(User.login)])
+                .select_from(join(Group, Membership).join(User))
+                .where(Membership.active())
+                .group_by(Group.id)
+                .as_scalar().label('members'))
+        .all()
+    )
```

Comment:
This has probably been discussed and I'm out of the loop, but what is the rationale behind circumventing the permission system of pycroft and syncing groups instead of properties?
For instance, it does not seem possible to me to block a user from certain services without terminating the group membership. (CC @sebschrader )

Reply:
Actually, both groups and properties are synced. But now that you mention it, I haven't thought whether we really need both, or if only syncing the properties would be enough. After looking at @sebschrader's config for OpenLDAP's SQL backend, I assumed that it was consensus that both groups and properties should be synced.